### PR TITLE
Adds gate blocks, for additional raid protection

### PR DESCRIPTION
**Pull request by someaddons**
# Changes proposed in this pull request:
This adds two gate blocks, one wooden one iron which are craftable by the mechanic.
Those have a max size of 5x4 and combine their hardness depending on size and act like vanilla doors, so that citizens and any other door using mechanic should work with them.

Fix raiders still using their old anti-stuck mechanics in addition to the new stuckhandler ones.
Fix herder trying to breed with one item.
Fix raid difficulty starting at zero

image
image

Review please

### CODE REVIEW

**Comment by Raycoms** [comment]
I'm not really happy having all the placement handlers in a different file, it was so easy to check and compare them in the old way =(

**Comment by someaddons** [comment]
It was terrible to find and compare them in a single file :D

**Comment by Raycoms** [comment]
they were all in a single location and all super small =P

**Comment by someaddons** [comment]
multiple files = side by side comparison,  single file = stupid scrolling up and down all the time :D besides the IDe search doesnt find subclasses that well

File: src/main/resources/assets/minecolonies/lang/en_us.json [hunk 1]
```diff
@@ -849,6 +853,9 @@
   "com.minecolonies.coremod.item.bannerrallyguards.gui.error": "Error modifying banner. Please try again.",
   "block.minecolonies.blockpaperwall": "Paper Wall",
 
+  "com.minecolonies.coremod.item.gate.craft": "Crafted by the mechanic",
+  "com.minecolonies.coremod.item.gate.place": "Autoplaces parts, max 5x4 which need a stack of 20",
```
Comment (ravenbuilder934) [inline src/main/resources/assets/minecolonies/lang/en_us.json]:
```suggestion
  "com.minecolonies.coremod.item.gate.place": "Autoplaces parts. The max size is 5x4 which needs a stack of 20.",
```

File: src/main/resources/assets/minecolonies/lang/en_us.json [hunk 2]
```diff
@@ -849,6 +853,9 @@
   "com.minecolonies.coremod.item.bannerrallyguards.gui.error": "Error modifying banner. Please try again.",
   "block.minecolonies.blockpaperwall": "Paper Wall",
 
+  "com.minecolonies.coremod.item.gate.craft": "Crafted by the mechanic",
```
Comment (ravenbuilder934) [inline src/main/resources/assets/minecolonies/lang/en_us.json]:
```suggestion
  "com.minecolonies.coremod.item.gate.craft": "Crafted by the Mechanic",
```

**Review by ravenbuilder934 — CHANGES_REQUESTED** [review]

**Review by marchermans — APPROVED** [review]
I approve of the splitting of the handlers. Also nice addition!

**Pull request merged**